assert_test.go

```diff
@@ -0,0 +1,55 @@
+package gosnowflake
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func assertNil(t *testing.T, actual interface{}, descriptions ...string) {
+	if actual != nil {
+		desc := joinDescriptions(descriptions...)
+		errMsg := fmt.Sprintf("expected \"%s\" to be nil but was not%s", actual, desc)
+		t.Fatal(errMsg)
```

Comment:
Do we need fatal here? Maybe error is better?

Reply:
I prefer fatal, because most likely I would like to break the test execution.
Common test scenario is to execute "when" operation and in the first order to check if there is error. Only if there is no error then we assert on the result. If there was an error and we got nil as the response and if we try assert something on nil as if it was a regular object we might get panic errors.
So we could have 2 versions of these methods: one for fatal, one for error, but for now for simplicity I did only the fatal one.

Reply:
Ok, so going with go way, what I suggest is naming functions like `assertNilF` and when it is needed we can add `assertNilE`? This way every attempt may be considered individually.

Reply:
I added both versions for each assert